packages/flutter_tools/test/dependency_checker_test.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter_tools/src/dart/dependencies.dart';
+import 'package:flutter_tools/src/dependency_checker.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+import 'src/common.dart';
+import 'src/context.dart';
+
+void main()  {
+  group('DependencyChecker', () {
+    final String basePath = path.dirname(Platform.script.path);
+    final String dataPath = path.join(basePath, 'data', 'dart_dependencies_test');
+    testUsingContext('good', () {
+      final String testPath = path.join(dataPath, 'good');
+      final String mainPath = path.join(testPath, 'main.dart');
+      final String fooPath = path.join(testPath, 'foo.dart');
+      final String barPath = path.join(testPath, 'lib', 'bar.dart');
+      final String packagesPath = path.join(testPath, '.packages');
+      DartDependencySetBuilder builder =
+          new DartDependencySetBuilder(mainPath, testPath, packagesPath);
+      DependencyChecker dependencyChecker =
+          new DependencyChecker(builder, null);
+
+      // Set file modification time on all dependencies to be in the past.
+      updateFileModificationTime(packagesPath, -10);
```

Comment:
Change updateModificationTime to take a `baseTime` and the offset, then record `now` as a local variable and pass  that as `baseTime`.  This'll save you from the (admittedly very very low) chance that the system clock will tick in the middle of this method and introduce flakiness.